### ISSUE TITLE
TSServer High CPU usage

### ISSUE THREAD

**Issue by kasik96** [Microsoft/vscode#70891]
- VSCode Version: 1.32.3
- OS Version: Fedora 29 

Steps to Reproduce:

1. Install clean stable VSCode with neo extensions or setting
2. Open large project with sub projects
3. Check System Monitor -> HIGH CPU Usage, almost non usable OS

I tried to turn off some settings, but nothing help.
It caused by "TypeScript Language Basics" built-in extension.

![Screenshot from 2019-03-21 11-18-59](https://user-images.githubusercontent.com/7235992/54746201-d7a6fc00-4bcb-11e9-8786-899c9e974a3e.png)
![Screenshot from 2019-03-21 11-19-27](https://user-images.githubusercontent.com/7235992/54746202-d7a6fc00-4bcb-11e9-81e1-975e4d673091.png)
![Screenshot from 2019-03-21 11-19-56](https://user-images.githubusercontent.com/7235992/54746203-d7a6fc00-4bcb-11e9-9bd3-40345783019d.png)

Solution: Disable "TypeScript Language Basics" built-in extension and Restart.

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all extensions disabled?

**Comment by kasik96** [comment]
@mjbvz  Yes :/ 
![Screenshot from 2019-03-21 19-20-16](https://user-images.githubusercontent.com/7235992/54775528-7eac8780-4c0e-11e9-8aa6-57b20299404b.png)

**Comment by kasik96** [comment]
Also i have the same problem with Ubuntu (Unity, XFCE), Mate and Fedora.

**Comment by kasik96** [comment]
@mjbvz  Any update on this bug? Also tested on Manjaro, i have same problem. It does not look it happens on Windows, it has little more CPU use, but not that much as in Linux.

**Comment by ZachGoldberg** [comment]
Same here.  Ubuntu 18.04.  Identical symptoms.  Disabling JS & TS features built-in also fixes it.

**Comment by abhishek-rs** [comment]
Have the same issue on MacOS Mojave. Using this as a workaround - 
1. Disable the default javascript validation - 
`"javascript.validate.enable": false`
2. Use the external ESLint extension - https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

**Comment by mjbvz** [comment]
@kasik96 Please try collecting the TS Server log:

1. Set `"typescript.tsserver.log": "verbose",`
1. Restart VS Code and reproduce the problem
1. In VS Code, run the `TypeScript: Open TS Server log` command
1. This should open a folder with a `tsserver.log` file in it

If you can share the log, I can also take a look to see if anything stands out that could be causing this

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by kasik96** [comment]
@mjbvz Yeah, no problem, here it is: https://gist.github.com/kasik96/0af966b0b28f7a118025e50f943429d8

**Comment by stefems** [comment]
+1
I'm having the same cpu issue with `eletron_node tsserver.js` as shown in the vs code process explorer. Pls fix soon. My laptop has been melting my fingers. Suggestion by abhishek-rs has been working thus far.

**Comment by npesa92** [comment]
+1 Having the same issue here. One line change will spike CPU  and show electron_node tsserver.js as the problem

**Comment by SeedyROM** [comment]
Why did I update!!! This is making VSCode unusable for me. Is there anyway to download and older version of VSCode in the mean time?

**Comment by SeedyROM** [comment]
+1 Can confirm on Ubuntu 16.04, problem seems to be stemming from tsserver. I have nearly identical logs to kasik96. Use and older version of VSCode instead of trying to hack this into working!
